jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -982,7 +983,23 @@ def _fit_model_to_cube(self, add_data):
 
         sb_unit = self.app._get_display_unit('sb')
         if spec.flux.unit != sb_unit:
-            spec = spec.with_flux_unit(sb_unit)
+            # ensure specutils has access to jdaviz custom unit equivalencies
+            pixar_sr = spec.meta.get('_pixel_scale_factor', None)
+            equivalencies = all_flux_unit_conversion_equivs(pixar_sr=pixar_sr,
+                                                            cube_wave=spec.spectral_axis)
+
+            pix2_in_flux = 'pix2' in spec.flux.unit.to_string()
+            pix2_in_sb = 'pix2' in sb_unit
+            # Handle various cases when PIX2 angle unit is present in the conversion
+            if pix2_in_flux and pix2_in_sb:
+                spec = spec.with_flux_unit(u.Unit(spec.flux.unit)*PIX2, equivalencies=equivalencies)  # noqa
+                spec = spec.with_flux_unit(u.Unit(sb_unit)*PIX2, equivalencies=equivalencies)
+            elif pix2_in_flux:
+                spec = spec.with_flux_unit(u.Unit(spec.flux.unit) * PIX2, equivalencies=equivalencies)  # noqa
+            elif pix2_in_sb:
+                spec = spec.with_flux_unit(u.Unit(spec.flux.unit) / PIX2, equivalencies=equivalencies)  # noqa
+
+            spec = spec.with_flux_unit(sb_unit, equivalencies=equivalencies)
```

Comment:
Maybe in a small effort before `flux_conversion` and `flux_conversion_general`, these both right now return the converted `values` as an array or  `spectral_axis`  as a Quantity object. It'd be worth adding the ability to return the `spec` Spectrum1D object as well and then we can update this logic block. 

@cshanahan1 if you test with this cube: 
`cubeviz.load_data(f"https://data.sdss.org/sas/dr17/manga/spectro/redux/v3_1_1/7443/stack/manga-7443-12703-LOGCUBE.fits.gz")`

and convert the flux_unit to Jy, this no longer throughs a UnitConversionError for the case you brought up. Test coverage is there now too! 
 
This PR is a blocker for a notebook we need merged once 4.1 is released next week, so I will create a ticket and start working on the Spectrum1D compatibility when converting flux.